src/System.Private.CoreLib/src/System/Variant.cs

```diff
@@ -112,180 +110,143 @@ internal struct Variant
         };
 
         internal static readonly Variant Empty = new Variant();
-        internal static readonly Variant Missing = new Variant(Variant.CV_MISSING, Type.Missing, 0, 0);
-        internal static readonly Variant DBNull = new Variant(Variant.CV_NULL, System.DBNull.Value, 0, 0);
+        internal static readonly Variant Missing = new Variant(Variant.CV_MISSING, Type.Missing);
+        internal static readonly Variant DBNull = new Variant(Variant.CV_NULL, System.DBNull.Value);
 
         //
         // Native Methods
         //
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal extern double GetR8FromVar();
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal extern float GetR4FromVar();
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal extern void SetFieldsR4(float val);
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal extern void SetFieldsR8(double val);
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
         internal extern void SetFieldsObject(object val);
 
-        // Use this function instead of an ECALL - saves about 150 clock cycles
-        // by avoiding the ecall transition and because the JIT inlines this.
-        // Ends up only taking about 1/8 the time of the ECALL version.
-        internal long GetI8FromVar()
-        {
-            return ((long)m_data2 << 32 | ((long)m_data1 & 0xFFFFFFFFL));
-        }
-
         //
         // Constructors
         //
 
-        internal Variant(int flags, object or, int data1, int data2)
+        internal Variant(int flags, object or, long data = 0)
```

Comment:
I don't see any usage of `Variant` that needs the default value for `data`. I would prefer if that was explicit.

Reply:
These two lines depend on the default value: https://github.com/dotnet/coreclr/pull/24951/files#diff-6493f97f2c623d9845d14c55ab8a6eafR113. I can change them to pass 0 explicitly ... I do not have a preference.

Reply:
Sorry, I meant existing lines not in this PR. Since those lines were already being touched I didn't count them. I would prefer them to be explicit in this case.